kani-driver/src/harness_runner.rs

```diff
@@ -101,6 +104,46 @@ impl<'sess, 'pr> HarnessRunner<'sess, 'pr> {
 }
 
 impl KaniSession {
+
+    fn process_output(&self, result: &VerificationResult, harness: &HarnessMetadata) {
+        if self.should_print_output() {
+            if self.args.output_into_files {
+                self.write_output_to_file(result, harness);
+            } 
+
+            let output = result.render(&self.args.output_format, harness.attributes.should_panic, self.args.coverage);
+            println!("{}", output);
+        }
+    }
+
+    fn should_print_output(&self) -> bool {
+        !self.args.common_args.quiet && self.args.output_format != OutputFormat::Old
+    }
+
+    fn write_output_to_file(&self, result: &VerificationResult, harness: &HarnessMetadata) {
+        let target_dir = self.get_target_dir();
+        let file_name = format!("{}/{}", target_dir, harness.pretty_name);
+        let path = Path::new(&file_name);
+        let prefix = path.parent().unwrap();
+
+        std::fs::create_dir_all(prefix).unwrap();
+        let mut file = File::create(&file_name).unwrap();
+        let file_output = result.render(&OutputFormat::Regular, harness.attributes.should_panic, self.args.coverage);
+
+        if let Err(e) = writeln!(file, "{}", file_output) {
+            eprintln!("Failed to write to file {}: {}", file_name, e);
+        }
+    }
+
+    fn get_target_dir(&self) -> String {
```

Comment:
Can you please rename this to `get_result_output_dir` to separate these files from the other artifacts that Kani may generate?